lib/elixir_sense/core/introspection.ex

```diff
@@ -606,19 +606,82 @@ defmodule ElixirSense.Core.Introspection do
     end
   end
 
-  def actual_mod_fun(mod_fun, imports, aliases, current_module) do
+  defp maybe_expand_alias(nil, _aliases), do: nil
+
+  defp maybe_expand_alias(mod, aliases) do
+    case Enum.find(aliases, fn {a, _m} -> a == mod end) do
+      nil -> mod
+      {_a, m} -> m
+    end
+  end
+
+  def actual_mod_fun({nil, nil}, _, _, _, _), do: {nil, nil}
+
+  def actual_mod_fun(mod_fun = {mod, fun}, imports, aliases, current_module, mods_funs) do
     with {nil, nil} <- find_kernel_function(mod_fun),
-         {nil, nil} <- find_imported_function(mod_fun, imports),
-         {nil, nil} <- find_aliased_function(mod_fun, aliases),
-         {nil, nil} <- find_function_in_module(mod_fun),
-         {nil, nil} <- find_builtin_type(mod_fun),
-         {nil, nil} <- find_function_in_current_module(mod_fun, current_module) do
+         {nil, nil} <-
+           find_metadata_function(
+             {maybe_expand_alias(mod, aliases), fun},
+             current_module,
+             imports,
+             mods_funs
+           ),
+         {nil, nil} <- find_builtin_type(mod_fun) do
       mod_fun
     else
       new_mod_fun -> new_mod_fun
     end
   end
 
+  defp has_type?(mod, type) do
+    ElixirSense.Core.Normalized.Typespec.get_types(mod)
+    |> Enum.any?(fn {_kind, {name, _def, _args}} -> name == type end)
+  end
+
+  defp find_metadata_function({nil, fun}, current_module, imports, mods_funs) do
+    mods = [current_module | imports]
+
+    case mods
+         |> Enum.find(fn mod ->
+           find_metadata_function({mod, fun}, current_module, imports, mods_funs) != {nil, nil}
+         end) do
+      nil -> {nil, nil}
+      mod -> {mod, fun}
+    end
+  end
+
+  defp find_metadata_function({mod, nil}, _current_module, _imports, mods_funs) do
+    case Code.ensure_loaded(mod) do
+      {:module, _} ->
+        {mod, nil}
+
+      _ ->
+        case mods_funs[mod] do
+          nil -> {nil, nil}
+          _ -> {mod, nil}
+        end
+    end
+  end
+
+  defp find_metadata_function({mod, fun}, _current_module, _imports, mods_funs) do
+    if ModuleInfo.has_function?(mod, fun) or has_type?(mod, fun) do
```

Comment:
Consider rewriting this as a cond

Reply:
I refactored it to prefer metadata over introspection and simplified it a bit